fern/ir-types-v30/definition/http.yml

```diff
@@ -0,0 +1,255 @@
+      name: commons.Name
+      properties: list<FileUploadRequestProperty>
+  BytesRequest:
+    properties:
+      isOptional: boolean
+      contentType: optional<string>
+  FileUploadRequestProperty:
+    union:
+      file: FileProperty
+      bodyProperty: InlinedRequestBodyProperty
+  FileProperty:
+    properties:
+      key: commons.NameAndWireValue
+      isOptional: boolean
+  HttpRequestBodyReference:
+    extends: commons.WithDocs
+    properties:
+      requestBodyType: types.TypeReference
+      contentType: optional<string>
+  SdkRequestBodyType:
+    union:
+      typeReference: HttpRequestBodyReference
+      bytes: BytesRequest
+  SdkRequest:
+    properties:
+      requestParameterName: commons.Name
+      shape: SdkRequestShape
+  SdkRequestShape:
+    union:
+      justRequestBody: SdkRequestBodyType
+      wrapper: SdkRequestWrapper
+  SdkRequestWrapper:
+    properties:
+      wrapperName: commons.Name
+      bodyKey: commons.Name
+
+  HttpResponse:
+    union:
+      json: JsonResponse
+      fileDownload: FileDownloadResponse
+      text: TextResponse
+      streaming: StreamingResponse
+
+  JsonResponse:
+    union:
+      response: JsonResponseBody
+      nestedPropertyAsResponse: JsonResponseBodyWithProperty
+
+  JsonResponseBody:
+    extends: commons.WithDocs
+    properties:
+      responseBodyType: types.TypeReference
+
+  JsonResponseBodyWithProperty:
+    extends: commons.WithDocs
+    properties:
+      responseBodyType: types.TypeReference
+      responseProperty:
+        docs: |
+          If set, the SDK will return this property from
+          the response, rather than the response itself.
+
+          This is particularly useful for JSON API structures
+          (e.g. configure 'data' to return 'response.data').
+        type: optional<types.ObjectProperty>
+
+  FileDownloadResponse:
+    extends: commons.WithDocs
+
+  TextResponse:
+    extends: commons.WithDocs
+
+  StreamingResponse:
+    extends: commons.WithDocs
+    properties:
+      dataEventType: StreamingResponseChunkType
+      terminator: optional<string>
+
+  StreamingResponseChunkType:
+    union:
+      json:
+        type: types.TypeReference
+        key: json
+      text: {}
+
+  ResponseErrors: list<ResponseError>
+  ResponseError:
+    extends: commons.WithDocs
+    properties:
+      error: errors.DeclaredErrorName
+
+  # examples
+
+  ExampleEndpointCall:
+    extends: commons.WithDocs
+    properties:
+      name: optional<commons.Name>
+      url: string
+      rootPathParameters: list<ExamplePathParameter>
+      servicePathParameters: list<ExamplePathParameter>
+      endpointPathParameters: list<ExamplePathParameter>
+      serviceHeaders: list<ExampleHeader>
+      endpointHeaders: list<ExampleHeader>
+      queryParameters: list<ExampleQueryParameter>
+      request: optional<ExampleRequestBody>
+      response: ExampleResponse
+
+  ExamplePathParameter:
+    properties:
+      name: commons.NameAndWireValue
```

Comment:
Path parameters don't _really_ have a wire representation - they're just placeholders for the value that's actually sent on the wire (e.g. `/user/{id}` => `/user/123`). However, we could use `commons.NameAndWireValue` here for consistency with the other endpoint parameters (e.g. query parameters and headers), and simply set the `wireValue` to the same thing as the `name`'s original value.

There might even be a use case later to have this representation of the "wire value", such as including the default path structure vs. another name used in the SDKs. For example, we might want to include `/user/{id}` somewhere, and treat `id` as the "wire value", but the example snippet might want to use a `userId` parameter name in the generated SDK signature (which would be the "name").

Alternatively, we just use `commons.Name` here and drop the wire value altogether.

Reply:
> Alternatively, we just use commons.Name here and drop the wire value altogether.

lets do this!